bio2zarr/vcf2zarr/vcz.py

```diff
@@ -240,993 +80,156 @@ def convert_local_allele_field_types(fields):
-            fixed_field_spec(
-                name="variant_allele",
-                dtype="O",
-                shape=(m, max_alleles),
-                dimensions=["variants", "alleles"],
-                chunks=(variants_chunk_size, max_alleles),
-            ),
-            fixed_field_spec(
-                name="variant_id",
-                dtype="O",
-            ),
-            fixed_field_spec(
-                name="variant_id_mask",
-                dtype="bool",
-            ),
-        ]
-        name_map = {field.full_name: field for field in icf.metadata.fields}
-
-        # Only three of the fixed fields have a direct one-to-one mapping.
-        array_specs.extend(
-            [
-                spec_from_field(name_map["QUAL"], array_name="variant_quality"),
-                spec_from_field(name_map["POS"], array_name="variant_position"),
-                spec_from_field(name_map["rlen"], array_name="variant_length"),
-            ]
-        )
-        array_specs.extend(
-            [spec_from_field(field) for field in icf.metadata.info_fields]
-        )
-
-        gt_field = None
-        for field in icf.metadata.format_fields:
-            if field.name == "GT":
-                gt_field = field
-                continue
-            array_specs.append(spec_from_field(field))
-
-        if gt_field is not None and n > 0:
-            ploidy = max(gt_field.summary.max_number - 1, 1)
-            shape = [m, n]
-            chunks = [variants_chunk_size, samples_chunk_size]
-            dimensions = ["variants", "samples"]
-            array_specs.append(
-                ZarrArraySpec.new(
-                    vcf_field=None,
-                    name="call_genotype_phased",
-                    dtype="bool",
-                    shape=list(shape),
-                    chunks=list(chunks),
-                    dimensions=list(dimensions),
-                    description="",
-                )
-            )
-            shape += [ploidy]
-            chunks += [ploidy]
-            dimensions += ["ploidy"]
-            array_specs.append(
-                ZarrArraySpec.new(
-                    vcf_field=None,
-                    name="call_genotype",
-                    dtype=gt_field.smallest_dtype(),
-                    shape=list(shape),
-                    chunks=list(chunks),
-                    dimensions=list(dimensions),
-                    description="",
-                )
-            )
-            array_specs.append(
-                ZarrArraySpec.new(
-                    vcf_field=None,
-                    name="call_genotype_mask",
-                    dtype="bool",
-                    shape=list(shape),
-                    chunks=list(chunks),
-                    dimensions=list(dimensions),
-                    description="",
-                )
-            )
-
-        if local_alleles:
-            array_specs = convert_local_allele_field_types(array_specs)
+def generate_schema(
```

Comment:
Shouldn't this just be a method of the ICF/Plink formats? The vcz module shouldn't be creating schemas I think.